psi4/driver/task_planner.py

```diff
@@ -0,0 +1,257 @@
+#
+# The copyrights for code used from other parties are included in
+# the corresponding files.
+#
+# This file is part of Psi4.
+#
+# Psi4 is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, version 3.
+#
+# Psi4 is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with Psi4; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# @END LICENSE
+#
+
+__all__ = ["task_planner"]
+
+import os
+import copy
+import logging
+from typing import Dict, Tuple, Union
+
+from qcelemental.models import DriverEnum
+from psi4.driver import p4util, pp
+from psi4.driver.task_base import AtomicComputer
+from psi4.driver.driver_findif import FiniteDifferenceComputer
+from psi4.driver.driver_nbody import ManyBodyComputer
+from psi4.driver.driver_cbs import CompositeComputer, composite_procedures, cbs_text_parser
+from psi4.driver.driver_util import negotiate_derivative_type, negotiate_convergence_criterion
+
+logger = logging.getLogger(__name__)
+
+TaskComputers = Union[AtomicComputer, CompositeComputer, FiniteDifferenceComputer, ManyBodyComputer]
+
+
+def _expand_cbs_methods(method: str, basis: str, driver: DriverEnum, **kwargs) -> Tuple[str, str, Dict]:
+    if method == 'cbs' and kwargs.get('cbsmeta', None):
+        return method, basis, kwargs['cbsmeta']
+
+    # Expand CBS methods
+    if "/" in method:
+        kwargs["ptype"] = driver
+        cbsmeta = cbs_text_parser(method, **kwargs)
+
+        # Single call detected
+        if "cbs_metadata" not in cbsmeta:
+            method = cbsmeta["method"]
+            basis = cbsmeta["basis"]
+        else:
+            method = "cbs"
+    else:
+        cbsmeta = {}
+
+    return method, basis, cbsmeta
+
+
+def task_planner(driver: DriverEnum, method: str, molecule: "psi4.core.Molecule", **kwargs) -> TaskComputers:
+    """Plans a task graph of a complex computation.
+
+    Canonical Task layering:
+     - ManyBody
+     - FiniteDifference
+     - Composite
+     - Atomic
+
+    Parameters
+    ----------
+    driver
+        The resulting type of computation: e/g/h. Note for finite difference that this should be the target driver, not the means driver.
+    method
+        A string representation of the method such as "HF" or "B3LYP". Special cases are:
+        "cbs"
+    molecule
+        A Psi4 base molecule to use
+    kwargs
+        User keyword arguments, often used to configure task computers.
+
+    Returns
+    -------
+    computer
+        A simple (AtomicComputer) or Layered (CompositeComputer, FiniteDifferenceComputer, ManyBodyComputer) task object.
```

Comment:
Is "Layered" the name of an abstract base class? Otherwise, I'm not sure why this is capitalized.

Reply:
nope, ordinary adjective. I'll fix the case.